File: wire-android-automation/common/src/main/java/com/wearezeta/auto/common/calling2/v1/model/InstanceRequest.java
package com.wearezeta.auto.common.calling2.v1.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@JsonIgnoreProperties(ignoreUnknown = true)
public final class InstanceRequest {

	private final String email;
	private final String password;
	private final String verificationCode;
	private final String backend;
	private final BackendDTO customBackend;
	private final VersionedInstanceType instanceType;
	private final String name;
	private final long timeout;
	private final boolean beta;

	public InstanceRequest(String email, String password, String verificationCode, String backend,
						   BackendDTO customBackend, VersionedInstanceType instanceType, String name, boolean beta,
						   long timeout) {
		this.email = email;
		this.password = password;
		this.verificationCode = verificationCode;
		this.backend = backend;
		this.customBackend = customBackend;
		this.instanceType = instanceType;
		this.name = name;
		this.beta = beta;
		this.timeout = timeout;
	}

	public String getEmail() {
		return email;
	}

	public String getPassword() {
		return password;
	}

	public String getVerificationCode() {
		return verificationCode;
	}

	public String getBackend() {
		return backend;
	}

	public BackendDTO getCustomBackend() {
		return customBackend;
	}

	public VersionedInstanceType getInstanceType() {
		return instanceType;
	}

	public long getTimeout() {
		return timeout;
	}

	public boolean getBeta() {
		return beta;
	}

	public String getName() {
		return name;
	}

	@Override
	public String toString() {
		return "InstanceRequest{email=" + email + ", password=" + password
				+ ", environment=" + backend + ", customBackend=" + customBackend.getBackendUrl() + " | " + customBackend.getWebappUrl() + ", instanceType=" + instanceType
				+ ", name=" + name + ", timeout=" + timeout + ", beta=" + beta +'}';
	}

}

File: tests/testsSupport/src/main/call/models/InstanceRequest.kt
/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 */
package call.models

data class InstanceRequest(
    val email: String,
    val password: String,
    val verificationCode: String? = null,
    val backend: String,
    val customBackend: BackendDTO,
    val instanceType: VersionedInstanceType,
    val name: String,
    val beta: Boolean,
    val timeout: Long
) {
    override fun toString(): String {
        return "InstanceRequest(email=$email, password=$password, environment=$backend, " +
                "customBackend=${customBackend.backendUrl} | ${customBackend.webappUrl}, " +
                "instanceType=$instanceType, name=$name, timeout=$timeout, beta=$beta)"
    }
}
